### ISSUE TITLE
Support for Azure managed identities?

### ISSUE THREAD

**Issue by fabiendelpierre** [acmesh-official/acme.sh#2912]
Hi there,

I'm using Azure DNS for the validation step, everything works fine, but I lament the fact that you require an Azure AD service principal in every situation.

I have an Azure VM with a [managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) that lets me interact with Azure APIs without having to manage credentials. I just do `az login --identity` and I'm automatically granted permissions according to a custom role I also defined and assigned to the managed identity. This is similar to using an [AWS IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html) to facilitate authentication to the API and is considered best practice.

This would tie in nicely with your own [suggested practice of limiting access to TXT records](https://github.com/acmesh-official/acme.sh/wiki/How-to-use-Azure-DNS#limit-access-permissions-to-txt-records), but unfortunately acme.sh refuses to run unless I set `AZUREDNS_APPID` and `AZUREDNS_CLIENTSECRET`, which I have to request from the team that has access to Azure AD to create service principals, who will then give me secrets I will have to keep safe. This could all be avoided simply by leveraging the managed identity.

tl;dr please consider adding support for Azure managed identities, it's easier and more secure.

**Comment by yaroslav-nakonechnikov** [comment]
yes, it works!

**Comment by neilpang** [comment]
Implemented and confirmed working in this thread: set `AZUREDNS_SUBSCRIPTIONID` and `AZUREDNS_MANAGEDIDENTITY=true` to use the VM's managed identity instead of a service principal (see https://github.com/acmesh-official/acme.sh/wiki/dnsapi#dns_azure). Azure Arc (non-Azure machines) is tracked separately in #4567. Closing as implemented.
